### PR TITLE
Updated the skeleton

### PR DESCRIPTION
**Pull request by dsahoo-nexient**
Added a variable enable_rbac_authorization
Wrote examples/complete
Wrote new terratests
Updated the skeleton
make check execution successful

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,9 +1,8 @@
...
-pre-commit 2.17.0
-terragrunt 0.36.1
+conftest 0.44.1
+golang 1.20.8
+golangci-lint 1.55.2
+pre-commit 3.3.3
+terraform 1.5.5
+terraform-docs 0.16.0
+terragrunt 0.39.2
+tflint 0.48.0
```
Comment (chris11-taylor-nttd) [inline .tool-versions:8]: Missing newline on this and other files, combined with pre-commit-config.yaml removal/add to .gitignore suggests that you need to `make configure` to pull the .pre-commit-config.yaml back in to your local and either `pre-commit run --all` or make another commit to force pre-commit to run again.

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,112 @@
-module github.com/nexient-llc/tf-azurerm-module-key_vault
+module github.com/nexient-llc/tf-azurerm-module_primitive-key_vault
```
Comment (chris11-taylor-nttd) [inline go.mod:1]: I'm surprised tests actually ran, this is supposed to be a valid URL and this module hasn't been renamed in GitHub yet. Are you planning to rename this repo as soon as this PR merges or will you need someone else to do that for you?

File: NOTICE [hunk 3]
```diff
@@ -0,0 +1,8 @@
+tf-azurerm-module_primitive-key_vault
+Copyright 2024 NTT Data, Inc.
```
Comment (chris11-taylor-nttd) [inline NOTICE:2]:
```suggestion
Copyright 2024 NTT DATA, Inc.
```

File: examples/complete/variables.tf [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+variable "enable_rbac_authorization" {
+  description = "Enable RBAC authorization for the key vault"
+  type        = bool
+  default     = false
+}
+
+# Variables to import pre existing certificates to the key vault
+variable "certificates" {
+  description = "List of certificates to be imported. The pfx files should be present in the root of the module (path.root) and its name denoted as certificate_name"
```
Comment (chris11-taylor-nttd) [inline examples/complete/variables.tf:107]: Given that this module is asking users to place their .pfx files in the root of the module, should we add .pfx files to the .gitignore to ensure nobody accidentally pushes them to source control?

We can add an override to allow a particular .pfx file to exist in an examples/ folder if we want to someday include that as part of an example (or test), but it seems like we'd want to put up some safeguards in the context of the root module.

File: go.mod (hunk 2 above)
Comment (dsahoo-nexient) [reply go.mod:1]: I will rename this repo. And as far the go tests goes, for self references go actually doesnt check the modules in the github. it picks up the packages locally.

File: examples/complete/variables.tf (hunk 4 above)
Comment (dsahoo-nexient) [reply examples/complete/variables.tf:107]: its already in .gitignore

File: .tool-versions (hunk 1 above)
Comment (dsahoo-nexient) [reply .tool-versions]: Thanks. looks like the pre-commit was never run the existing code.

**Review by nexient-ajoshi — APPROVED** [review]
